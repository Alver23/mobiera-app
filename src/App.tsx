// Core configuration
import './core/icon';

// Dependencies
import React, { ReactElement } from 'react';
import { ThemeProvider, Text } from 'react-native-elements';
import { SafeAreaView, StatusBar, useColorScheme, View } from 'react-native';

import { Colors } from 'react-native/Libraries/NewAppScreen';

// @Core
import { theme } from './core/theme';

const App = (): ReactElement => {
  const isDarkMode = useColorScheme() === 'dark';

  const backgroundStyle = {
    backgroundColor: isDarkMode ? Colors.darker : Colors.lighter,
  };

  return (
    <ThemeProvider theme={theme}>
      <SafeAreaView style={backgroundStyle}>
        <StatusBar barStyle={isDarkMode ? 'light-content' : 'dark-content'} />
        <View>
          <Text>App Mobiera</Text>
        </View>
      </SafeAreaView>
    </ThemeProvider>
  );
};

export default App;
