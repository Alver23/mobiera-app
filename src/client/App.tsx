// Core configuration
import 'reflect-metadata';
import 'es6-shim';
import './core/icon';
import './core/tools/reactotron';

// Dependencies
import 'react-native-gesture-handler';
import React, { ReactElement } from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'react-native-elements';
import { SafeAreaView, useColorScheme } from 'react-native';
import { Colors } from 'react-native/Libraries/NewAppScreen';

// Components
import StatusBar from '@mobiera/components/status-bar';

// Containers
import AuthProvider from '@mobiera/containers/auth-provider';
import AppContainer from '@mobiera/containers/app';

// Store
import store from './store';

// @Core
import { theme } from './core/theme';

// Styles
import useStyles from './styles';

const App = (): ReactElement => {
  const isDarkMode = useColorScheme() === 'dark';
  const containerBackgroundColor = isDarkMode ? Colors.darker : Colors.lighter;
  const styles = useStyles({ containerBackgroundColor });

  return (
    <Provider store={store}>
      <ThemeProvider theme={theme} useDark={isDarkMode}>
        <AuthProvider>
          <SafeAreaView style={styles.container}>
            <StatusBar />
            <AppContainer />
          </SafeAreaView>
        </AuthProvider>
      </ThemeProvider>
    </Provider>
  );
};

export default App;
