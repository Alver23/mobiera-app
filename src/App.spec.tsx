// Dependencies
import React from 'react';
import { shallow, ShallowRendererProps } from 'enzyme';

// Containers
import App from '@mobiera/App';

// Mocks
jest.mock('./store', () => {
  const store = {
    getState: jest.fn(() => ({})),
    dispatch: jest.fn(),
  };

  return store;
});
jest.mock('react-redux', () => ({
  Provider: jest.fn(),
}));
jest.mock('@mobiera/containers/app', () => jest.fn());
jest.mock('@mobiera/containers/auth-provider', () => jest.fn());
jest.mock('./core/tools/reactotron', () => jest.fn());

describe('App Component', () => {
  let component: ShallowRendererProps;

  beforeEach(() => {
    component = shallow(<App />);
  });

  it('should save a snapshot of the component', () => {
    expect(component).toMatchSnapshot();
  });
});
