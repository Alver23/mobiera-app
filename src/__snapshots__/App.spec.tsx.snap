// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component should save a snapshot of the component 1`] = `
<mockConstructor
  store={
    Object {
      "dispatch": [MockFunction],
      "getState": [MockFunction],
    }
  }
>
  <ThemeProvider
    theme={
      Object {
        "Button": Object {
          "buttonStyle": Object {
            "height": 50,
            "width": 312,
          },
          "containerStyle": Object {
            "alignItems": "center",
          },
          "titleStyle": Object {
            "fontFamily": "Helvetica",
            "fontSize": 17,
          },
        },
        "Input": Object {
          "containerStyle": Object {
            "alignItems": "center",
          },
          "inputContainerStyle": Array [
            Object {
              "backgroundColor": "#ffffff",
              "borderRadius": 5,
              "borderWidth": 2,
              "height": 48,
              "paddingHorizontal": 10,
              "width": 312,
            },
          ],
          "inputStyle": Object {
            "fontSize": 17,
          },
        },
        "Text": Object {
          "h4Style": Object {
            "fontSize": 30,
          },
          "style": Object {
            "color": "#ffffff",
            "fontFamily": "Helvetica",
            "fontSize": 15,
          },
        },
      }
    }
    useDark={false}
  >
    <mockConstructor>
      <ForwardRef(SafeAreaView)
        style={
          Object {
            "backgroundColor": "#F3F3F3",
            "flex": 1,
          }
        }
      >
        <StatusBarComponent />
        <mockConstructor />
      </ForwardRef(SafeAreaView)>
    </mockConstructor>
  </ThemeProvider>
</mockConstructor>
`;
